Sources/MapboxMaps/Style/Style.swift

```diff
@@ -798,21 +784,27 @@ public final class Style: StyleProtocol {
         return styleManager.getStyleLightProperty(forProperty: property)
     }
 
-    /// Sets a value to the style light property.
+    // MARK: - Terrain
+
+    /// Sets a terrain on the style
     ///
-    /// - Parameters:
-    ///   - property: Style light property name.
-    ///   - value: Style light property value.
+    /// - Parameter terrain: The `Terrain` that should be rendered
     ///
     /// - Throws:
     ///     An error describing why the operation was unsuccessful.
-    public func setLightProperty(_ property: String, value: Any) throws {
-        return try handleExpected {
-            return styleManager.setStyleLightPropertyForProperty(property, value: value)
+    public func setTerrain(_ terrain: Terrain) throws {
+        let terrainData = try JSONEncoder().encode(terrain)
+        guard let terrainDictionary = try JSONSerialization.jsonObject(with: terrainData) as? [String: Any] else {
+            throw TypeConversionError.unexpectedType
         }
+
+        try setTerrain(properties: terrainDictionary)
     }
 
-    // MARK: - Terrain
+    /// Removes terrain from style if it was set.
+    public func removeTerrain() {
+        styleManager.setStyleTerrainForProperties(NSNull())
+    }
```

Comment:
This is the interesting part. The rest of the changes to this file are just reorganizing light and terrain methods so that they're all next to each other. Previously, there were separate groups for high-level and low-level terrain and light methods.